animeal/src/Networking/NetworkRequest.swift

```diff
@@ -0,0 +1,141 @@
+import AWSPluginsCore
+import Amplify
+
+struct Request<R: Decodable> {
+    let apiName: String?
+    let document: String
+    let variables: [String: Any]?
+    let responseType: R.Type
+    let decodePath: String?
+
+    init(
+        apiName: String? = nil,
```

Comment:
nit: Maybe we can also write parameters without default values first and then parameters with default values?

Reply:
Done